### ISSUE TITLE
Hostname/IP validation is too strict

### ISSUE THREAD

**Issue by RaphiMC** [PaperMC/Velocity#1031]
If I want to use `raphimc.net_25564_1.8.x.viaproxy.127-0-0-1.nip.io:25568` as a hostname for my backend server in the velocity config it complains that the address is not valid. Pinging `raphimc.net_25564_1.8.x.viaproxy.127-0-0-1.nip.io` in cmd resolves to localhost as it should. The reason I want to use such a long hostname is to pass on information (Through the address in the handshake packet) to the "backend server" (Another proxy) to know where to connect to. Trying to connect to that address with a minecraft client works fine.

**Comment by RaphiMC** [comment]
Bungeecord has the same issue: https://github.com/SpigotMC/BungeeCord/issues/3480

**Comment by electronicboy** [comment]
I mean, you're breaking the hostname spec, I guess we probably should relax this a little, but, idk what extra chars we allow people to break the spec with

**Comment by RaphiMC** [comment]
Which of the characters breaks the hostname spec?

**Comment by electronicboy** [comment]
_

**Comment by RaphiMC** [comment]
Just tried, even browsers accept _ in the URLs. So I think it should be relaxed even tho its not strictly per spec

**Comment by electronicboy** [comment]
Yes, that's what I'm saying, I just have no idea what extra characters we
allow

**Comment by samuelh2005** [comment]
~~Even Docker Swarm uses `_` in container hostnames if a stack name is used - and there is no way to change that.~~ - Actualy that is just the container name, and all containers of a service just share one hostname anyway *(you would have to use  the IP to identify a single container)*

> Yes, that's what I'm saying, I just have no idea what extra characters we
> allow

I'm thinking just to allow `_` as they seem to quite common.
